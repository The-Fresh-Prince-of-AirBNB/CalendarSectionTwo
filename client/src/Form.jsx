import React, { useState, useEffect } from 'react';
import Calendar from './Calendar.jsx';
import Guests from './Guests.jsx';

const Form = (props) => {
  const [displayCalendar, setDisplayCalendar] = useState(false);
  const [displayGuests, setDisplayGuests] = useState(false);
  const [x, setX] = useState(0);
  const [y, setY] = useState(0);

  const toggleCalendar = () => (
    setDisplayCalendar(!displayCalendar)
  );

  const toggleGuests = () => {
    setDisplayGuests(!displayGuests);
  };

  const mousePosition = (event) => {
    setX(event.clientX);
    setY(event.clientY);
  };

  return (
    <div>
      <div className="form">
        <div>
          <button className="check" onClick={() => toggleCalendar()} type="button">
            <div className="checkIn">
              <div style={{ fontSize: '12px' }}>CHECK-IN</div>
              <div style={{ color: 'rgb(125, 125, 125)' }}>Add date</div>
            </div>
            <div className="checkOut">
              <div style={{ fontSize: '12px' }}>CHECKOUT</div>
              <div style={{ color: 'rgb(125, 125, 125)' }}>Add date</div>
            </div>
          </button>
        </div>
        <div style={{ display: displayCalendar ? 'block' : 'none' }}>
          <Calendar
            close={toggleCalendar}
            setFees={props.setFees}
            setDates={props.setDates}
            dates={props.dates}
            fees={props.fees}
          />
        </div>
        <div>
          <button className="guestBut" onClick={() => toggleGuests()} type="button">
            <div style={{ fontSize: '12px' }}>GUESTS</div>
            <div>1 guest</div>
          </button>
          <div style={{ display: displayGuests ? 'block' : 'none' }}>
            <Guests close={toggleGuests} />
          </div>
        </div>
      </div>
      <div>
        <button
          className="checkAvailability"
          type="button"
          style={{ backgroundPosition: `${x}px ${y}px`, backgroundImage: `linear-gradient(to right, #d43939, #d43974)` }}
          onMouseMove={(event) => mousePosition(event)}
          onMouseLeave={() => {
            setX(0);
            setY(0);
          }}
        >
          Check Availability
        </button>
      </div>
    </div>
  );
};

export default Form;
