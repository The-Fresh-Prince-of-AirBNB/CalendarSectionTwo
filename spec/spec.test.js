/* eslint-disable no-undef */

import React from 'react';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Form from './../client/src/Form.jsx';

configure({ adapter: new Adapter() });

describe('testing for jest', () => {
  it('test to see that jest runs', () => {
    // Common Matching
    const data = { one: 1 };
    data.two = 2;
    expect(data).toEqual({ one: 1, two: 2 });
    expect(1 + 2).toBe(3);

    // Truthiness
    const n = null;
    expect(n).toBeNull();
  });
});

describe('Snapshot in debug mode', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Form debug />);
    expect(component).toMatchSnapshot();
  });
});
