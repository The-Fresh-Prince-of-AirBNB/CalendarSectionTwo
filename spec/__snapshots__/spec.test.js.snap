// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render correctly in "debug" mode 1`] = `
<div>
  <div
    className="form"
  >
    <div>
      <button
        className="check"
        onClick={[Function]}
        type="button"
      >
        <div
          className="checkIn"
        >
          <div
            style={
              Object {
                "fontSize": "12px",
              }
            }
          >
            CHECK-IN
          </div>
          <div
            style={
              Object {
                "color": "rgb(125, 125, 125)",
              }
            }
          >
            Add date
          </div>
        </div>
        <div
          className="checkOut"
        >
          <div
            style={
              Object {
                "fontSize": "12px",
              }
            }
          >
            CHECKOUT
          </div>
          <div
            style={
              Object {
                "color": "rgb(125, 125, 125)",
              }
            }
          >
            Add date
          </div>
        </div>
      </button>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <Calendar
        close={[Function]}
        dateSet={
          Object {
            "debug": true,
          }
        }
      />
    </div>
    <div>
      <button
        className="guestBut"
        onClick={[Function]}
        type="button"
      >
        <div
          style={
            Object {
              "fontSize": "12px",
            }
          }
        >
          GUESTS
        </div>
        <div>
          1 guest
        </div>
      </button>
      <div
        style={
          Object {
            "display": "none",
          }
        }
      >
        <Guests
          close={[Function]}
        />
      </div>
    </div>
  </div>
  <div>
    <button
      className="checkAvailability"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      style={
        Object {
          "backgroundImage": "linear-gradient(to right, #d43939, #d43974)",
          "backgroundPosition": "0px 0px",
        }
      }
      type="button"
    >
      Check Availability
    </button>
  </div>
</div>
`;
